cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataSubscriberServiceTest.java

```diff
@@ -190,6 +199,58 @@ public void testWorkflow() throws InterruptedException, ExecutionException, Time
     }
   }
 
+  @Test
+  public void testMetadata() throws InterruptedException, TimeoutException, ExecutionException {
+    ProgramRunId workflowRunId = workflow1.run(RunIds.generate());
+    MetadataEntity entity = MetadataEntity.ofDataset("myns", "myds");
+
+    // Try to read, should have nothing
+    MetadataStore metadataStore = getInjector().getInstance(MetadataStore.class);
+    MetadataRecordV2 meta = metadataStore.getMetadata(MetadataScope.USER, entity);
+    Assert.assertTrue(meta.getProperties().isEmpty());
+    Assert.assertTrue(meta.getTags().isEmpty());
+
+    // Start the MetadataSubscriberService
+    MetadataWriter metadataWriter = getInjector().getInstance(MessagingMetadataWriter.class);
+    MetadataSubscriberService subscriberService = getInjector().getInstance(MetadataSubscriberService.class);
+    subscriberService.startAndWait();
+    try {
+
+      // publish metadata put
+      Map<String, String> propertiesToAdd = ImmutableMap.of("a", "x", "b", "z");
+      Set<String> tagsToAdd = ImmutableSet.of("t1", "t2");
+      metadataWriter.add(workflowRunId, entity, propertiesToAdd, tagsToAdd);
+
+      // wait until meta data is written
+      Tasks.waitFor(true, () -> {
+            MetadataRecordV2 record = metadataStore.getMetadata(MetadataScope.USER, entity);
+            return !record.getProperties().isEmpty() && !record.getTags().isEmpty();
+          }, 10, TimeUnit.SECONDS, 100, TimeUnit.MILLISECONDS);
+
+      // validate correctness of meta data written
+      meta = metadataStore.getMetadata(MetadataScope.USER, entity);
+      Assert.assertEquals(propertiesToAdd, meta.getProperties());
+      Assert.assertEquals(tagsToAdd, meta.getTags());
+
+      // publish metadata delete
+      metadataWriter.remove(workflowRunId, entity, ImmutableSet.of("a"), ImmutableSet.of("t1", "t3"));
+
+      // wait until meta data is written
+      Tasks.waitFor(true, () -> {
+        MetadataRecordV2 record = metadataStore.getMetadata(MetadataScope.USER, entity);
```

Comment:
This is a different indentation than above. Please make it consistent (I like the indentation here more).

Reply:
^This is what IntelliJ does on my Linux desktop...